### PR TITLE
update to new Bresch decoder from main repo

### LINKED ISSUE #1277: update of obs2ioda
## Description

the NCAR obs2ioda has fallen behind and should be brought up to date so other repairs and fixes can be brought in and routine maintenance can be performed

### PR DESCRIPTION
**Pull request by PatNichols**
## Description

This PR will merge changes from the Jamie Bresch obs2ioda branch into ioda converters.

## Issue(s) addressed

Resolves #1277 

## Impact

No expected impacts to jedi codes.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by PatNichols** [comment]
> I am working on adding "Launchtime" to radiosonde data. will commit a little later.

thanks @hailingz !

**Comment by climbfuji** [comment]
@PatNichols I just pushed commit https://github.com/JCSDA-internal/ioda-converters/pull/1279/commits/b7d0c584661594f8fe92267159c6e5ea7c82994b, which fixes a compiler error on AWS Parallel Cluster with `intel@2021.6.0`:
```
/usr/bin/ld: failed to convert GOTPCREL relocation; relink with –no-relax
```

**Comment by PatNichols** [comment]
> @PatNichols I just pushed commit [b7d0c58](https://github.com/JCSDA-internal/ioda-converters/commit/b7d0c584661594f8fe92267159c6e5ea7c82994b), which fixes a compiler error on AWS Parallel Cluster with `intel@2021.6.0`:
> 
> ```
> /usr/bin/ld: failed to convert GOTPCREL relocation; relink with –no-relax
> ```

**Comment by BenjaminRuston** [comment]
@hailingz you push now has it changing 855 files!

I had already tried to put in your prepBUFR changes already:
https://github.com/JCSDA-internal/ioda-converters/pull/1279/commits/2569c1c1abbb852a389c888de80773a44560efcf

note that has commit I did previously

**Comment by hailingz** [comment]
OOPS, I did not mean it. how could I reverse my commit then? @BenjaminRuston 
I saw your changes but there is one missing.

**Comment by PatNichols** [comment]
@hailingz Just let me know what needs to be changed. There isn't much change from the ncar obs2ioda so it shouldn't be a big deal to change this.

**Comment by hailingz** [comment]
> @hailingz Just let me know what needs to be changed. There isn't much change from the ncar obs2ioda so it shouldn't be a big deal to change this.

sorry for messing up. 
The only change I meant to make is replace line 187 with
if (ncname == 'dateTime' .or. ncname == 'LaunchTime') then

Thanks!

**Comment by BenjaminRuston** [comment]
@hailingz yeah I don't think this was your change not sure what's up

**Comment by hailingz** [comment]
> @hailingz yeah I don't think this was your change not sure what's up

I do not know what happened. Did what I usually do to commit and push :(

**Comment by PatNichols** [comment]
> > @hailingz yeah I don't think this was your change not sure what's up
> 
> I do not know what happened. Did what I usually do to commit and push :(

@hailingz I had to revert your changes. Every file in the tree except ncio_mod were deleted :)

**Comment by PatNichols** [comment]
@hailingz @BenjaminRuston Fixed with revert and changed to ncio_mod to what it should be?
Also changed the ncio_mod from *.h5 to *.nc4 ( I think that's right).

File: cmake/compiler_flags_GNU_Fortran.cmake [hunk 1]
```diff
@@ -8,25 +8,25 @@
 # FLAGS COMMON TO ALL BUILD TYPES
 ####################################################################
 
-set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -ffree-line-length-none")
+set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -g -ffree-line-length-none -mcmodel=medium")
```
Comment (BenjaminRuston) [inline cmake/compiler_flags_GNU_Fortran.cmake:11]: think the `mcmodel` is a Intel compiler flag not `gnu` I do not see it in the gnu flags

Comment (PatNichols) [reply cmake/compiler_flags_GNU_Fortran.cmake:11]: @BenjaminRuston it's a GNU flag. Both intel and gnu used the same backend up until recently. Intel is moving to clang (may have already done so with OneAPI). It's a cpu dependent  flag for arm and x86.
https://gcc.gnu.org/onlinedocs/gcc/Option-Summary.html

Comment (climbfuji) [reply cmake/compiler_flags_GNU_Fortran.cmake:11]: @PatNichols You probably saw that pushed those flags for Intel only directly in the CMakeLists in the obs2ioda subdirectory. Feel free to take those out and make them consistent with the gnu flags, or vice versa. Thanks!

**Comment by PatNichols** [comment]
There is a gotcha for the clang build. Since we have to use the fortran compiler from gnu, the "-mcmodel=medium" flags causes a problem for a  linking library built with clang since they cannot be linked due to the conflicting memory models.

File: cmake/compiler_flags_GNU_Fortran.cmake (hunk 1 above)
Comment (PatNichols) [reply cmake/compiler_flags_GNU_Fortran.cmake:11]: @climbfuji Thanks! I have migrated those changes into our cmake/compiler_flags.

**Comment by PatNichols** [comment]
> There is a gotcha for the clang build. Since we have to use the fortran compiler from gnu, the "-mcmodel=medium" flags causes a problem for a linking library built with clang since they cannot be linked due to the conflicting memory models.

This is fixed now. We should at some point separate compiler flags for macos apple clang and clang on linux which are not the same system.

**Comment by PatNichols** [comment]
This PR still needs an updated testoutput file to compare to for ctests.

**Comment by PatNichols** [comment]
@BenjaminRuston  got rid of the mopitt file.

**Pull request closed without merging**